### PR TITLE
[WIP] Create deployment infrastructure for fuzzers and other services.

### PR DESCRIPTION
**Pull request by WhiteEyeDoll**
###### Scripts and tools

| Name | Language | Features | Status |
| --- | --- | --- | --- |
| lifufi-client | Python 3 | Small client for CorpusCloud | Under work |
###### Kubernetes deployments

| Target | Status |
| --- | --- |
| lua | Under work |
| libxml2 | Not implemented |
###### Infra

| Service |  Type | Features | Components | Status |
| --- | --- | --- | --- | --- |
| CorpusShare | p2p | Distributes corpus directories via p2p | Deluge, Google container engine & cloud storage | Under work |
| CorpusCloud | web | Web app for managing corpus distribution | Google app engine & cloud storage | Not implemented |
| Homepage | web |  | Github pages | Initial template ready at: lifufi.ouspg.org |

### CODE REVIEW

File: docker/infra/share-p2p/nodeconfig.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+# This script does the following:
+# 1. Configures the Deluge daemon and starts it.
+# 2. Gets info about the most current libfuzzerfication sample collection(s)
+#    and optionally downloads the file(s) directly from a resource location.
+#    (for example a local http/ftp server for a better performance or
+#    when there isn't existing seeds)
+# 3. Adds the torrent(s) from a magnet link and starts seeding them.
+
+funtion help () {
```
Comment (oherrala) [inline docker/infra/share-p2p/nodeconfig.sh]: Systematic typo in this file. "funtion" should be "function"?

File: docker/stubs/ImageMagick/ImageMagick_fuzzer.c [hunk 2]
```diff
@@ -4,26 +4,27 @@
 #include <string.h>
 #include "MagickCore/MagickCore.h"
 
-/*
-  We use BlobToImage to load input as an image, if successful destroy image.  
-*/
+# We use BlobToImage to load input as an image, if successful destroy image.
```
Comment (oherrala) [inline docker/stubs/ImageMagick/ImageMagick_fuzzer.c]: This is not valid C comment. The original `/* */` syntax is valid. `#` is used mostly by preprocessor.

File: kubernetes/scripts/corpus-util-demo.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+function download () {
+    curl -O --output $sampledir $CORPUS_SERVICE_URL
+    curl -O --output $sampledir $CORPUS_URL/$SHA256SUMS_FILE
+}
+
+funtion upload () {
+    echo "Funtion does not exist."
+}
+
+funtion check () {
```
Comment (oherrala) [inline kubernetes/scripts/corpus-util-demo.sh]: `funtion` should be `function`?

File: docker/infra/share-p2p/nodeconfig.sh (hunk 1 above)
Comment (WhiteEyeDoll) [inline docker/infra/share-p2p/nodeconfig.sh]: Fixed in 802ab4b9b493f164eb31788647040ff59b2fa086

File: docker/stubs/ImageMagick/ImageMagick_fuzzer.c (hunk 2 above)
Comment (WhiteEyeDoll) [inline docker/stubs/ImageMagick/ImageMagick_fuzzer.c]: Fixed in 46f67b5836505c838632d1606765f8e4859557e4

File: docker/infra/share-p2p/nodeconfig.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    help
+}
+
+
+
+sampledir="/srv/deluge/samples"
+autoadd_dir="/srv/deluge/autoadd"
+baseurl="http://storage.googleapis.com/libfuzzerfication/samples"
+user=admin
+passwd=libfuzzerfication
```
Comment (oherrala) [inline docker/infra/share-p2p/nodeconfig.sh]: Passwords look really bad in source code and public repository. Read the credentials (user name + password) from file which is not version controlled in repository.

File: kubernetes/scripts/corpus-util-demo.sh (hunk 3 above)
Comment (WhiteEyeDoll) [inline kubernetes/scripts/corpus-util-demo.sh]: Fixen in 1196c8c878cb512e142920d44bab8c334dc7772c

File: docker/infra/share-p2p/nodeconfig.sh (hunk 4 above)
Comment (WhiteEyeDoll) [inline docker/infra/share-p2p/nodeconfig.sh]: Fixed in 3536d009a534554a6c3dc411bbca6f2c2e3de5fa

**Pull request closed without merging**